modules/internal/ExternalString.chpl

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module ExternalString {
+  use CPtr;
+
+  //
+  // Use this as a convenient way to get a handle to "char*" during resolution.
+  //
+  type chpl__exportTypeCharPtr = c_ptr(c_char);
+
+  //
+  // TODO: Look at `modules/internal/String.chpl` to make this copy faster.
+  // There is no point in calling `calloc` here when each byte is going to
```

Comment:
Might as well just change it to malloc now if you know this.

Reply:
The problem is that I'm not sure how to set the NULL terminator to zero given just a `c_ptr(c_char)`. I'll have to see what the `string` implementation does.